### PR TITLE
feat: add risky role checks

### LINKED ISSUE #670: feat: rbac assessment rego policies
Create rego policies that will perform risky role assessments in the cluster.
depend on

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>
Add risky role checks for role and role binding
close #670

### CODE REVIEW

**Review by liamg — CHANGES_REQUESTED** [review]

File: internal/rules/rbac/allowing_create_role_binding_and_associate_privileged_clusterrole.rego [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	input.rules[_].apiGroups[_] == "rbac.authorization.k8s.io"
+	input.rules[_].resources[_] == ["clusterroles", "roles"][_]
+	input.rules[_].verbs[_] == "bind"
+	input.rules[_].resourceNames[_] == "*"
+}
+
+deny[res] {
+	allowing_create_role_binding_and_associate_cluster_role
+	msg := "Role permits creation of role binding and association with privileged role"
+	res := result.new(msg, input)
```
[the remark this answers is not in the document]
Comment (chen-keinan) [reply internal/rules/rbac/allowing_create_role_binding_and_associate_privileged_clusterrole.rego:39]: In this case, the problem when the above combination of both rules are met , so , should I return both ?

[the remark this answers is not in the document]
Comment (liamg) [reply internal/rules/rbac/allowing_create_role_binding_and_associate_privileged_clusterrole.rego:39]: I think something like this should solve the problem - the `i < j` stops the duplicate result being reported:
```rego
allowing_create_role_binding_and_associate_cluster_role[ruleA] {
	input.kind == readKinds[_]
	some i, j
	ruleA := input.rules[i]
	ruleB := input.rules[j]
	i < j
	ruleA.apiGroups[_] == "rbac.authorization.k8s.io"
	ruleA.resources[_] == "rolebindings"
	ruleA.verbs[_] == "create"
	ruleA.apiGroups[_] == "rbac.authorization.k8s.io"
	ruleB.resources[_] == ["clusterroles", "roles"][_]
	ruleB.verbs[_] == "bind"
	ruleB.resourceNames[_] == "*"
	ruleB.apiGroups[_] == "rbac.authorization.k8s.io"
}

deny[res] {
	badRule := allowing_create_role_binding_and_associate_cluster_role[_]
	msg := "Role permits creation of role binding and association with privileged role"
	res := result.new(msg, badRule)
}
```

[the remark this answers is not in the document]
Comment (liamg) [reply internal/rules/rbac/allowing_create_role_binding_and_associate_privileged_clusterrole.rego:39]: We currently only support highlighting a single map, so passing one of the two offending rules should be enough to point the user in the right direction.

File: internal/rules/rbac/allowing_create_role_binding_and_associate_privileged_clusterrole_test.rego [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+			{
+				"apiGroups": ["rbac.authorization.k8s.io"],
+				"resources": ["clusterroles"],
+				"verbs": ["bind"],
+				"resourceNames": ["*"],
+			},
+		],
+	}
+
+	count(r) > 0
```
Comment (liamg) [inline internal/rules/rbac/allowing_create_role_binding_and_associate_privileged_clusterrole_test.rego:26]: I think we should use `count(r) == 1` in these tests, as we want to make sure duplicates aren't reported to the user.

**Review by liamg — CHANGES_REQUESTED** [review]

**Review by liamg — APPROVED** [review]
LGTM :rocket:

**Pull request merged**